### PR TITLE
feat: support vfolder trash bin

### LINKED ISSUE #2202: Let's introduce trash bin to the vfolder list page.
### Main idea

From https://github.com/lablup/backend.ai/pull/835 PR, we can use vfolder trash bin.
But there's not corresponding PR for UI. So, the `DELETE_ONGOING` and `DELETE_COMPLETE` status vfolders are displayed.
Let's add a trash bin button or trash bin tab to the vfolder list and filter the `DELETE_ONGOING`, `DELETE_COMPLETE` status vfolders.

### Alternative ideas

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by agatha197**
resolves #2202 
core related PR: https://github.com/lablup/backend.ai/pull/1892

## Checklist for reviewers.
- [ ] Only `delete-pending` status folders are controllable.
image
- [ ] Other folders whose status is one of the [DEAD_VFOLDER_STATUSES](https://github.com/lablup/backend.ai/blob/cd3b660886dd3aa685b755cf025cccffa1fc5588/src/ai/backend/manager/models/vfolder.py#L182) except for `delete-pending` are disabled. But you can check the info.
image
- [ ] Vfolders that belong to the `DEAD_VFOLDER_STATUSES` are listed only in the trash bin tab.
- [ ] If you delete the folder, it will be moved to the trash bin tab. And the status is `delete-pending`.
- [ ] If you click the restore button, the folder status will be changed to `ready`.
- [ ] If you click the delete_forever button, the folder status will be changed to `delete-ongoing` -> `delete-complete` / `delete-error`. 
- [ ] If the folder status is `delete-complete`, you cannot see the folder info in UI. (you can check with DB).
- [ ] Can you create a new folder with the same name as the deleted forever folder?

## How to test
move this line to line number 677.

https://github.com/lablup/backend.ai-webui/blob/c1b3bb4a73f52d9c2228b44d65fef9f30ee84ee5/src/lib/backend.ai-client-esm.ts#L680

**Checklist:** (if applicable)

- [x] Mention to the original issue
- [ ] Documentation
- [x] Minimum required manager version: 24.03.0*
- [x] Specific setting for review (eg., KB link, endpoint or how to setup)
- [x] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Comment by yomybaby** [comment]
I'll review this PR after merging https://github.com/lablup/backend.ai/pull/1892 @agatha197

File: src/components/backend-ai-storage-list.ts [hunk 1]
```diff
@@ -1544,6 +1567,24 @@ export default class BackendAiStorageList extends BackendAIPage {
...
+        </div>
+        <div
+          slot="footer"
+          class="horizontal end-justified flex layout"
+          style="gap:5px;"
+        >
+          <mwc-button outlined @click="${(e) => this._hideDialog(e)}">
+            ${_t('button.Cancel')}
+          </mwc-button>
+          <mwc-button raised @click="${() => this._deleteForever()}">
```
Comment (yomybaby) [inline src/components/backend-ai-storage-list.ts]: image

Please change the color of "영구삭제" button to destroy color.

File: src/components/backend-ai-storage-list.ts [hunk 2]
```diff
@@ -4376,6 +4500,50 @@ export default class BackendAiStorageList extends BackendAIPage {
...
+          this.notification.detail = err.message;
+          this.notification.show(true, err);
+        }
+      });
+  }
+
+  /**
+   * Call `delete_from_trash_bin` API to delete the folder permanently.
+   * */
+  _deleteForever() {
```
Comment (yomybaby) [inline src/components/backend-ai-storage-list.ts]: How about matching the names related to "deleteForever" to API name "delete-from-trash-bin" ? `deteFromTrashBin`

**Review by yomybaby — CHANGES_REQUESTED** [review]

**Comment by yomybaby** [comment]
image

image

Since supporting the trash bin, the 'delete from trash bin' action needs to be confirmed by the user.

**Comment by agatha197** [comment]
> image image
> Since supporting the trash bin, the 'delete from trash bin' action needs to be confirmed by the user.

Thank you for your feedback! I modified as below :)
image

**Comment by yomybaby** [comment]
@agatha197 Please resolve conflicts.

**Comment by agatha197** [comment]
> kimsujin (@agatha197) Please resolve conflicts.

Done!

**Review by yomybaby — CHANGES_REQUESTED** [review]
If manager supports `vfolder-trash-bin`: 

- Tooltip: "Delete" to "Move to trash bin"
    image

- The confirmation modal below is not needed. However, the notification message should describe "휴지통으로 이동되었습니다."
    image
    image

**Review by yomybaby — CHANGES_REQUESTED** [review]
- How about changing the trash bin icon color to the normal color since "move to trash bin" is not a dangerous action?
image

**Comment by yomybaby** [comment]
@agatha197  Thank you for the update. It's better than before. Could you improve the messages to include the folder name?

- "휴지통으로 이동되었습니다." to "OOO 폴더가 휴지통으로 이동되었습니다."
- "폴더가 복원되었습니다." to " OOO 폴더가 복원되었습니다."
- "폴더가 완전히 삭제되었습니다." to "OOO 폴더가 완전히 삭제되었습니다."

**Review by yomybaby — APPROVED** [review]
LGTM

**Pull request merged**